src/StackInterface.php

```diff
@@ -1,11 +1,11 @@
 <?php
 
-namespace Psr\Http\Middleware;
+namespace Idealo\Middleware;
 
-use Psr\Http\Message\RequestInterface;
-use Psr\Http\Message\ResponseInterface;
+use Interop\Http\ServerMiddleware\DelegateInterface;
+use Interop\Http\ServerMiddleware\MiddlewareInterface;
 
-interface StackInterface
+interface StackInterface extends DelegateInterface
 {
     /**
      * Return an instance with the specified middleware added to the stack.
```

Comment:
i would just stay by current implementation a just change methods ```withMiddleware/withoutMiddleware``` visibility to ```private``` and get rid  of ```StackInterface```. what do you think?